### PR TITLE
Frontend comments functionality

### PR DESCRIPTION
**Pull request by eeels22**
# Changes proposed in this pull request
## Frontend comments

- add a comment to a post
- delete a comment on a post
- Note: this early implementation does not check that the deleter was the author of the post

## User Story
`As a user of Cuddly,`  
`I want to give a comment about a post,`
`so that I can share my opinion`

### CODE REVIEW

File: frontend/src/pages/posts/PostCard.jsx [hunk 1]
```diff
@@ -1,14 +1,56 @@
...
+  useEffect(() => {
+    CommentsApi.getAllComments(postId)
+      .then(({ data }) => setComments(data))
+      .catch((err) => console.error(err));
+  }, [setComments, postId]);
+
   return (
     <div>
       <div>
         <p>{post.contentText}</p>
```
Comment (NeuralAlchemist) [inline frontend/src/pages/posts/PostCard.jsx]: all good, but can remove extra line

File: frontend/src/pages/posts/comments/CommentCard.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  // Methods
+  const handleDelete = () => {
+    // Invoke the passed in event callback
+    onDeleteClick(postId, comment);
+  };
+
+  return (
+    <div>
+      <p>{comment.contentText}</p>
+
```
Comment (NeuralAlchemist) [inline frontend/src/pages/posts/comments/CommentCard.jsx]: looks good, remove extra line please

**Review by NeuralAlchemist — COMMENTED** [review]
looks great!

File: frontend/src/pages/posts/comments/CommentCard.jsx (hunk 2 above)
Comment (eeels22) [reply frontend/src/pages/posts/comments/CommentCard.jsx]: done, thanks

File: frontend/src/pages/posts/PostCard.jsx (hunk 1 above)
Comment (eeels22) [reply frontend/src/pages/posts/PostCard.jsx]: done thanks

**Review by NeuralAlchemist — APPROVED** [review]
Thanks!

File: frontend/src/pages/posts/comments/CommentCard.jsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+// NPM Packages
+import React from "react";
+
+export default function Comment({ postId, comment, onDeleteClick }) {
```
Comment (janko101) [inline frontend/src/pages/posts/comments/CommentCard.jsx]: Should a component name also be CommentCard?

File: frontend/src/api/CommentsApi.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+class CommentsApi {
+    getAllComments(postId) {
+        return Api.get('/posts/' + postId + '/comments');
+    }
+
+    createComment(postId, comment) {
+        return Api.post('/posts/' + postId + '/comments', comment);
+    }
+
+    updateComment(postId, commentId, comment) {
```
Comment (janko101) [inline frontend/src/api/CommentsApi.js:12]: To update should be put req

**Review by janko101 — COMMENTED** [review]
Did you implement update comment somewhere?

File: frontend/src/pages/posts/comments/CommentCard.jsx (hunk 3 above)
Comment (eeels22) [reply frontend/src/pages/posts/comments/CommentCard.jsx]: yes, good spot, thanks

File: frontend/src/api/CommentsApi.js (hunk 4 above)
Comment (eeels22) [reply frontend/src/api/CommentsApi.js:12]: great spot, thanks

File: frontend/src/pages/posts/comments/CommentCard.jsx (hunk 3 above)
Comment (eeels22) [reply frontend/src/pages/posts/comments/CommentCard.jsx]: done

File: frontend/src/api/CommentsApi.js (hunk 4 above)
Comment (eeels22) [reply frontend/src/api/CommentsApi.js:12]: done

**Review by janko101 — APPROVED** [review]

**Pull request merged**